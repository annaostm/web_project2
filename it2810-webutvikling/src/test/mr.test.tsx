import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import MR from '../components/mr/mr'

let container: Element | DocumentFragment | null = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  if(container == null){
    throw "container is null"
  }
  unmountComponentAtNode(container);
  container = null;
});

const b = renderer.create(<MR data={[{ merged: true, name: "noe", id: "1234", title: "issue", created_at: "221100", state: "state" }]} fromDateFilter={"271220220"} toDateFilter={"271220221"} stateFilter={""}></MR>);

it('renders correctly', () => {
    const tree= b.toJSON();
    expect(tree).toMatchSnapshot();
  });