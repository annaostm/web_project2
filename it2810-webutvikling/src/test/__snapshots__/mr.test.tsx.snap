// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Merge requests
  </h1>
  <p>
    No merge requests to show.
  </p>
</div>
`;



